runtime/bindings/python/src/pyopenvino/core/infer_request.cpp

```diff
@@ -51,19 +57,18 @@ void regclass_InferRequest(py::module m) {
         py::arg("inputs"));
 
     cls.def(
-        "_infer",
+        "infer",
         [](InferRequestWrapper& self, const py::dict& inputs) {
             // Update inputs if there are any
             if (!inputs.empty()) {
-                if (py::isinstance<py::str>(inputs.begin()->first)) {
-                    auto inputs_map = Common::cast_to_tensor_name_map(inputs);
-                    for (auto&& input : inputs_map) {
-                        self._request.set_tensor(input.first, input.second);
-                    }
-                } else if (py::isinstance<py::int_>(inputs.begin()->first)) {
-                    auto inputs_map = Common::cast_to_tensor_index_map(inputs);
-                    for (auto&& input : inputs_map) {
-                        self._request.set_input_tensor(input.first, input.second);
+                for (auto&& input : inputs) {
+                    if (py::isinstance<py::str>(input.first)) {
+                        self._request.set_tensor(input.first.cast<std::string>(), Common::cast_to_tensor(input.second));
+                    } else if (py::isinstance<py::int_>(input.first)) {
+                        self._request.set_input_tensor(input.first.cast<size_t>(),
+                                                       Common::cast_to_tensor(input.second));
+                    } else {
+                        throw ov::Exception("Incompatible key type for tensor!");
```

Comment:
doesn't it double-check?
you checked whether ```key``` is ```int``` or ```string``` in the ```ie_api.py```

Reply:
Discussed offline.